src/lib.rs

```diff
@@ -34,33 +34,32 @@ use serde::Serialize;
 
 /// A point in time.
 ///
-/// Low overhead time representation. Cannot be negative. Internally represented
-/// as milliseconds.
+/// Low overhead time representation. Internally represented as milliseconds.
 #[derive(
     Eq, PartialEq, Hash, Ord, PartialOrd, Copy, Clone, Debug, Default, Serialize, Deref, From, Into,
 )]
-pub struct Time(usize);
+pub struct Time(i64);
 impl Time {
-    pub const MAX: Self = Self(usize::MAX);
-    pub const EPOCH: Self = Self(0_usize);
+    pub const MAX: Self = Self(i64::MAX);
+    pub const EPOCH: Self = Self(0);
```

Comment:
We probably want to either set Time::default() to Time::EPOCH, or set Time::EPOCH to i64::MIN. Otherwise a lot of test cases in route-optimizer will likely fail.